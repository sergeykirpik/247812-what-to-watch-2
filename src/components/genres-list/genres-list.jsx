import React from 'react';
import PropTypes from 'prop-types';

const handleSelectItem = (e, genre, onSelectItem) => {
  e.preventDefault();
  onSelectItem(genre);
};

export const GenresList = (props) => {
  const {genres, activeItem, onSelectItem} = props;
  return (
    <ul className="catalog__genres-list">
      {genres.map((genre) => (
        <li key={genre} className={`catalog__genres-item ${genre.toUpperCase() === (activeItem || `All genres`).toUpperCase() && `catalog__genres-item--active`}`}>
          <a onClick={(e) => handleSelectItem(e, genre, onSelectItem)} href="#" className="catalog__genres-link">{genre}</a>
        </li>
      ))}
    </ul>
  );
};

GenresList.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  activeItem: PropTypes.string,
  onSelectItem: PropTypes.func.isRequired,
};

GenresList.defaultProps = {
  activeItem: `All genres`
};

export default GenresList;
