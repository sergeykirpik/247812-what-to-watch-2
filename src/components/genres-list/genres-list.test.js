import React from 'react';
import renderer from 'react-test-renderer';
import {GenresList} from './genres-list.jsx';

const genres = [`All genres`, `Comedy`, `Drama`];

it(`renders correctly`, () => {
  const tree = renderer
    .create(<GenresList genres={genres} onSelectItem={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
