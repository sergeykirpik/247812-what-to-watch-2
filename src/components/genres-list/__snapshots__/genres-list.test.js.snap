// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
</ul>
`;
